### PR TITLE
ZW Aselo Connector Twilio Configurations

### PR DESCRIPTION
**Pull request by sinehome**
Adding Aselo Connector studio flow updates on ZW terraform.

### CODE REVIEW

**Comment by sinehome** [comment]
ZW Staging Plan
- (Not sure about the 2 Post Survey Bot  AWS parameter additions)
- The 2 new task queue for Harare & Bulawayo
- Updating the Master workflow with new expression filters for each connector's host name ID

Terraform used the selected providers to generate the following execution
plan. Resource actions are indicated with the following symbols:
  + create
  ~ update in-place

Terraform will perform the following actions:

  # module.aws.aws_ssm_parameter.main_group["POST_SURVEY_BOT_CHAT_URL"] will be created
  + resource "aws_ssm_parameter" "main_group" {
      + arn            = (known after apply)
      + data_type      = (known after apply)
      + description    = (sensitive value)
      + id             = (known after apply)
      + insecure_value = (known after apply)
      + key_id         = (known after apply)
      + name           = (sensitive value)
      + tags_all       = (known after apply)
      + tier           = (known after apply)
      + type           = "SecureString"
      + value          = (sensitive value)
      + version        = (known after apply)
    }

  # module.aws.aws_ssm_parameter.twilio_post_survey_bot_chat_url[0] will be created
  + resource "aws_ssm_parameter" "twilio_post_survey_bot_chat_url" {
      + arn            = (known after apply)
      + data_type      = (known after apply)
      + description    = "Twilio account - Post Survey Bot Chat URL"
      + id             = (known after apply)
      + insecure_value = (known after apply)
      + key_id         = (known after apply)
      + name           = "/staging/twilio/AC48d146ce2460184b8944cc7fdf8c5d25/post_survey_bot_chat_url"
      + tags           = {
          + "Environment" = "staging"
          + "Name"        = "/staging/twilio/AC48d146ce2460184b8944cc7fdf8c5d25/post_survey_bot_chat_url"
          + "Terraform"   = "true"
        }
      + tags_all       = {
          + "Environment" = "staging"
          + "Name"        = "/staging/twilio/AC48d146ce2460184b8944cc7fdf8c5d25/post_survey_bot_chat_url"
          + "Terraform"   = "true"
        }
      + tier           = (known after apply)
      + type           = "SecureString"
      + value          = (sensitive value)
      + version        = (known after apply)
    }

  # module.taskRouter.twilio_taskrouter_workspaces_task_queues_v1.task_queue["bulawayo"] will be created
  + resource "twilio_taskrouter_workspaces_task_queues_v1" "task_queue" {
      + assignment_activity_sid  = (known after apply)
      + friendly_name            = "Bulawayo"
      + id                       = (known after apply)
      + max_reserved_workers     = (known after apply)
      + reservation_activity_sid = (known after apply)
      + sid                      = (known after apply)
      + target_workers           = "ac_hostname=='196.27.102.1'"
      + task_order               = (known after apply)
      + workspace_sid            = "WS1e1ed4a5e1f6875acdadda3c27b75ce8"
    }

  # module.taskRouter.twilio_taskrouter_workspaces_task_queues_v1.task_queue["harare"] will be created
  + resource "twilio_taskrouter_workspaces_task_queues_v1" "task_queue" {
      + assignment_activity_sid  = (known after apply)
      + friendly_name            = "Harare"
      + id                       = (known after apply)
      + max_reserved_workers     = (known after apply)
      + reservation_activity_sid = (known after apply)
      + sid                      = (known after apply)
      + target_workers           = "ac_hostname=='196.27.102.1'"
      + task_order               = (known after apply)
      + workspace_sid            = "WS1e1ed4a5e1f6875acdadda3c27b75ce8"
    }

  # module.taskRouter.twilio_taskrouter_workspaces_workflows_v1.workflow["master"] will be updated in-place
  ~ resource "twilio_taskrouter_workspaces_workflows_v1" "workflow" {
      ~ configuration            = jsonencode(
          ~ {
              ~ task_routing = {
                  ~ filters = [
                      + {
                          + expression           = "channelType =='web'  OR isContactlessTask == true OR  twilioNumber == 'messenger:103260519220529' OR aseloHostId == 'zw1' OR aseloHostId == 'zw2'"
                          + filter_friendly_name = "Childline Zimbabwe"
                          + targets              = [
                              + {
                                  + expression = "(worker.waitingOfflineContact != true AND ((task.channelType == 'voice' AND worker.channel.chat.assigned_tasks == 0) OR (task.channelType != 'voice' AND worker.channel.voice.assigned_tasks == 0)) AND ((task.transferTargetType == 'worker' AND task.targetSid == worker.sid) OR (task.transferTargetType != 'worker' AND worker.sid != task.ignoreAgent))) OR (worker.waitingOfflineContact == true AND task.targetSid == worker.sid AND task.isContactlessTask == true)"
                                  + queue      = "WQfdb52a16ea3a40371df56fbded3eec98"
                                },
                            ]
                        },
                        {
                            expression           = "channelType=='web' AND e2eTestMode=='true'"
                            filter_friendly_name = "E2E Test"
                            targets              = [
                                {
                                    expression = "(worker.waitingOfflineContact != true AND ((task.channelType == 'voice' AND worker.channel.chat.assigned_tasks == 0) OR (task.channelType != 'voice' AND worker.channel.voice.assigned_tasks == 0)) AND ((task.transferTargetType == 'worker' AND task.targetSid == worker.sid) OR (task.transferTargetType != 'worker' AND worker.sid != task.ignoreAgent))) OR (worker.waitingOfflineContact == true AND task.targetSid == worker.sid AND task.isContactlessTask == true)"
                                    queue      = "WQ35bdd78d4ba6ec970cfdbb7c1aa51403"
                                },
                            ]
                        },
                      - {
                          - expression           = "channelType =='web'  OR isContactlessTask == true OR  twilioNumber == 'messenger:103260519220529'"
                          - filter_friendly_name = "Childline Zimbabwe"
                          - targets              = [
                              - {
                                  - expression = "(worker.waitingOfflineContact != true AND ((task.channelType == 'voice' AND worker.channel.chat.assigned_tasks == 0) OR (task.channelType != 'voice' AND worker.channel.voice.assigned_tasks == 0)) AND ((task.transferTargetType == 'worker' AND task.targetSid == worker.sid) OR (task.transferTargetType != 'worker' AND worker.sid != task.ignoreAgent))) OR (worker.waitingOfflineContact == true AND task.targetSid == worker.sid AND task.isContactlessTask == true)"
                                  - queue      = "WQfdb52a16ea3a40371df56fbded3eec98"
                                },
                            ]
                        },
                    ]
                }
            }
        )
        id                       = "WS1e1ed4a5e1f6875acdadda3c27b75ce8/WWfd9eb357c72a412fb1853721f38c80b8"
        # (4 unchanged attributes hidden)
    }

  # module.taskRouter.twilio_taskrouter_workspaces_workflows_v1.workflow["survey"] will be updated in-place
  ~ resource "twilio_taskrouter_workspaces_workflows_v1" "workflow" {
      ~ configuration            = jsonencode( # whitespace changes
            {
                task_routing = {
                    filters = [
                        {
                            expression           = "isChatCaptureControl==true"
                            filter_friendly_name = "CaptureChannel"
                            targets              = [
                                {
                                    queue = "WQ97145e0faba1380e8788209c83184d1b"
                                },
                            ]
                        },
                    ]
                }
            }
        )
        id                       = "WS1e1ed4a5e1f6875acdadda3c27b75ce8/WW3a0b4342e65f519f5aa3e874aaa60298"
        # (4 unchanged attributes hidden)
    }

Plan: 4 to add, 2 to change, 0 to destroy.

**Comment by dee-luo** [comment]
@sinehome I made some updates to the TaskRouter filters to send the AC calls to distinct Harare and Bulawayo queues. We can always updates this later if Best would like the calls routed a different way, but I've also set it up so that workers will belong to their appropriate queues based on the Okta attributes. Just remember to update the Bulawayo hostname in the Task Queue worker expression, as well as update the acHostName attribute of the Bulawayo users in Okta once you have that info. 

Can you run another Plan? And then let's go ahead and apply the changes to Staging.

**Review by dee-luo — APPROVED** [review]

File: twilio-iac/helplines/zw/common.hcl [hunk 1]
```diff
@@ -30,6 +30,14 @@ locals {
         "target_workers" = "1==1",
         "friendly_name"  = "Childline Zimbabwe"
       },
+      harare : {
+        "target_workers" = "ac_hostname=='196.27.102.1'",
```
Comment (nickhurlburt) [inline twilio-iac/helplines/zw/common.hcl:34]: I don't love having this out in the open. It's fine for now, though. I'm not sure what the current state of the art is on whether we can hide these behind variables easily

File: twilio-iac/helplines/zw/common.hcl [hunk 2]
```diff
@@ -30,6 +30,14 @@ locals {
...
+        "friendly_name"  = "Harare"
+      },
+      bulawayo : {
+        "target_workers" = "ac_hostname=='196.27.102.1'",
```
Comment (nickhurlburt) [inline twilio-iac/helplines/zw/common.hcl:38]: And of course as soon as we know the correct IP for this, we should change it.

**Review by nickhurlburt — APPROVED** [review]
I did only a brief scan, but this looks right to me.

**Pull request merged**